__tests__/app.test.js

```diff
@@ -27,46 +27,119 @@ describe("GET /api/topics endpoint", () => {
+      });
+  });
+
+  test("Responds with a 404 error when the id passed is valid but no article", () => {
+    return request(app)
+      .get(`/api/articles/49532`)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toEqual("Resource not found");
+      });
+  });
+});
+
+describe("GET /api/users responds with an array of users", () => {
+  test("Responds with an array of users", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        let users = body;
+        users.forEach((user) => {
+          expect(user).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String),
             })
-          });
-    })
-    test('Responds with an empty object when the id passed is valid but has no article', () => {
-        const article_id = 2292943;
-        return request(app)
-        .get(`/api/articles/${article_id}`)
-        .expect(200)
-        .then((response) => {
-            expect(response.body).toEqual({});
-          });
-    })
-    test('Responds with a 404 error  when the id passed is invalid', () => {
-        return request(app)
-        .get(`/api/articles/notAnId`)
-        .expect(400)
-        .then((response) => {
-            console.log(response.body);
-            expect(response.body).toEqual({"msg": "Bad Request"});
-          });
-    })
-})
+          );
+        });
+        expect(users.length).toEqual(4);
+      });
+  });
+  test("Responds with 404 error when passed bad path", () => {
+    return request(app)
+    .get("/api/notapath")
+    .expect(404)
+  });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  test("200: Patch returns the updated article", () => {
+    const article_id = 2;
+    return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .send({ inc_votes: 1 })
+      .expect(200)
+      .then(({ body }) => {
+        expect(typeof body).toBe("object");
+      });
+  });
+  test("Newvote updates the votes property", () => {
+    const article_id = 2;
+    return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .send({ inc_votes: 1 })
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.votes).toBe(1);
+      });
+  });
+
+  test("Newvote updates the votes property", () => {
+    const article_id = 2;
+    return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .send({ inc_votes: -100 })
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.votes).toBe(-100);
+      });
+  });
+});
+
+test("404: Returns error message when article ID doesnt exist", () => {
+  const article_id = 99999;
+  return request(app)
+    .patch(`/api/articles/${article_id}`)
+    .send({ inc_votes: 1 })
+    .expect(404)
+    .then(({ body }) => {
+      expect(body.message).toBe("Resource not found");
+    });
```

Comment:
Tests need expanding, what about an invalid body? Missing parts or invalid inc amount? Plus needs to test for invalid ID